### ISSUE TITLE
Reusing CommandLine object doesn't reset values in argument groups options.

### ISSUE THREAD

**Issue by valepakh** [remkop/picocli#1768]
The following test fails:
```java
    @Test
    public void testReuseBothCommandLineAndUserObjectWithArgGroup() {
        Command userObject = new Command();
        CommandLine cmdLine = new CommandLine(userObject);
        cmdLine.execute("--group", "group", "--option", "option");
        cmdLine.execute();
        assertNull(userObject.option, "Expected option to be reset to null");
        assertNull(userObject.group.option, "Expected group option to be reset to null");
    }

    @Command(name = "command")
    static class Command implements Runnable {
        @Option(names = "--option")
        private String option;
        
        @ArgGroup
        private Group group;

        private static class Group {
            @Option(names = "--group")
            private String option;
        }

        @Override
        public void run() {
        }
    }
 ```

**Comment by remkop** [comment]
Apologies for the late reply.
For the option in the ArgGroup, you can force the value to be reset to `null` by specifying `defaultValue = Option.NULL_VALUE` in the annotation:

```java
    @Command(name = "command")
    static class MyCommand implements Runnable {
        @Option(names = "--option"/*, defaultValue = Option.NULL_VALUE*/)
        private String option;

        @ArgGroup
        private Group group = new Group();

        private static class Group {
            @Option(names = "--group", defaultValue = Option.NULL_VALUE)
            private String option;
        }

        public void run() {
        }
    }
```

**Comment by valepakh** [comment]
I might be missing something but the test still fails even with the `defaultValue = Option.NULL_VALUE` set.

**Comment by remkop** [comment]
I added [your test to `ArgGroupTest.java`](https://github.com/remkop/picocli/blob/main/src/test/java/picocli/ArgGroupTest.java#L4355).
I did have to switch the order of parameters in the `assertNull` statements because this test uses JUnit 4.
This test passes. Can you take a look?

**Comment by valepakh** [comment]
It looks like some fix from 4.7.0 fixed it, thanks!

**Comment by valepakh** [comment]
And it actually passes without `defaultValue = Option.NULL_VALUE`
